### ISSUE TITLE
About the loss function

### ISSUE THREAD

**Issue by moliniao** [ziqi-jin/finetune-anything#16]
Hello, can you explain the design principle of the loss function?

**Comment by ziqi-jin** [comment]
> Hello, can you explain the design principle of the loss function?

Recently I am editing [how to use](https://github.com/ziqi-jin/finetune-anything/blob/main/how_to_use_finetune_anything.md#losses), once I finish the introduction of the loss part, I will let you know in this issue

**Comment by moliniao** [comment]
But seeing that the implementation of the loss function in your code has been changing, what is the correct implementation?

**Comment by ziqi-jin** [comment]
> 



> But seeing that the implementation of the loss function in your code has been changing, what is the correct implementation?

Now the FA only support the losses of torch, you can add torch loss into this [name_dict](https://github.com/ziqi-jin/finetune-anything/blob/350c1fbf7f122a8525e7ffdecc40f259b262983f/losses/__init__.py#L4) and the get_loss function will get it when you give the related loss name in config file.

in the config file , for example CrossEntropy loss as below, first the loss name is `ce`, than the `weight` is used to set weight in total loss, the `params` should set the nesasarry params of `nn.CrossEntropyLoss`, after these settings above, you can use the torch losses.
```yaml
losses
    ce:
      weight: 0.5
      params:  # ~ means None type, the initial params of loss could be identified here
        ignore_index: 255
      label_one_hot: False
```

if you want to identify your own loss, I recommend create it in [loss.py](https://github.com/ziqi-jin/finetune-anything/blob/main/losses/losses.py),  implement the `__init__` and `forward` function, then add it in the name_dict. you are welcom to submit PR for a customized loss.

**Comment by ziqi-jin** [comment]
the loss part of [how to use](https://github.com/ziqi-jin/finetune-anything/blob/main/how_to_use_finetune_anything.md#losses) is updated.
